rcl_action/CMakeLists.txt

```diff
@@ -33,23 +33,25 @@ add_executable(test_compile_headers
 
 set(rcl_action_sources
   src/${PROJECT_NAME}/goal_state_machine.c
+  src/${PROJECT_NAME}/types.c
 )
 
 set_source_files_properties(
   ${rcl_action_sources}
   PROPERTIES language "C"
 )
 
-add_library(
-  ${PROJECT_NAME}
+add_library(${PROJECT_NAME}
   ${rcl_action_sources}
 )
+target_link_libraries(${PROJECT_NAME}
+  ${rcl_LIBRARIES}
```

Comment:
I think `ament_target_dependencies("rcl"...)` below does this already

https://github.com/ament/ament_cmake/blob/88add8cab0ca579e89e95b3ab343af319c163195/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L55